flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -3288,21 +3288,82 @@ IntrinsicLibrary::genSize(mlir::Type resultType,
       .getResults()[0];
 }
 
+static bool hasDefaultLowerBound(const fir::ExtendedValue &exv) {
+  return exv.match(
+      [](const fir::ArrayBoxValue &arr) { return arr.getLBounds().empty(); },
+      [](const fir::CharArrayBoxValue &arr) {
+        return arr.getLBounds().empty();
+      },
+      [](const fir::BoxValue &arr) { return arr.getLBounds().empty(); },
+      [](const auto &) { return false; });
+}
+
+/// Compute the lower bound in dimension \p dim (zero based) of \p array
+/// taking care of returning one when the related extent is zero.
+static mlir::Value computeLBOUND(fir::FirOpBuilder &builder, mlir::Location loc,
+                                 const fir::ExtendedValue &array, unsigned dim,
+                                 mlir::Value zero, mlir::Value one) {
+  assert(0 <= dim && dim < array.rank() && "invalid dimension");
```

Comment:
I get a warning on this line, which becomes an error since `-Werror` is enabled.  The problem is that `dim` is unsigned, which means that it's always >= 0, so the first expression in the assert is always true.

I've verified that if you remove the `0 <= dim && `, that all builds and tests correctly.

Reply:
Good catch, I removed it.